packages/@aws-cdk/core/lib/validation/report.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Violation produced by the validation plugin.
+ */
+export interface ValidationViolation {
+  /**
+   * The name of the rule.
+   */
+  readonly ruleName: string;
+
+  /**
+   * The recommendation to resolve the violation.
+   */
+  readonly recommendation: string;
+
+  /**
+   * How to fix the recommendation.
+   *
+   * @default - no fix is provided
+   */
+  readonly fix?: string;
+}
+
+/**
+ * Resource violating a specific rule.
+ */
+export interface ValidationViolatingResource {
+  /**
+   * The resource name.
+   */
+  readonly resourceName: string;
+
+  /**
+   * The locations in the CloudFormation template that pose the violations.
+   */
+  readonly locations: string[];
+
+  /**
+   * The path to the CloudFormation template that contains this resource
+   */
+  readonly templatePath: string;
+}
+
+/**
+ * Construct violating a specific rule.
+ */
+export interface ValidationViolatingConstruct extends ValidationViolatingResource {
+  /**
+   * The construct path as defined in the application.
+   */
+  readonly constructPath: string;
+
+  /**
+   * A stack of constructs that lead to the violation.
+   *
+   * @default - stack will be empty if the cli is not run with `--debug`
+   */
+  readonly constructStack?: string;
+}
+
+/**
+ * Validation produced by the validation plugin, in construct terms.
+ */
+export interface ValidationViolationConstructAware extends ValidationViolation {
+  /**
+   * The constructs violating this rule.
+   */
+  readonly violatingConstructs: ValidationViolatingConstruct[];
+}
+
+/**
+ * Validation produced by the validation plugin, in CFN resource terms
+ */
+export interface ValidationViolationResourceAware extends ValidationViolation {
+  /**
+   * The resources violating this rule.
+   */
+  readonly violatingResources: ValidationViolatingResource[];
+}
+
+
+/**
+ * The final status of the validation report
+ */
+export enum ValidationReportStatus {
+  /**
+   * No violations were found
+   */
+  SUCCESS = 'success',
+
+  /**
+   * At least one violation was found
+   */
+  FAILURE = 'failure',
+}
+
+/**
+ * Summary of the report.
+ */
+export interface ValidationReportSummary {
+  /**
+   * The final status of the validation (pass/fail)
+   */
+  readonly status: ValidationReportStatus;
+
+  /**
+   * The name of the plugin that created the report
+   */
+  readonly pluginName: string;
```

Comment:
Important to  show in the report output?

Reply:
We have considered this, but the problem is that plugin authors would have to keep this property in sync with the actual version of the package.

Reply:
Its important because different plugin versions might behave very differently...for example with respect to exemptions. 
Yeah I don't know how to solve this, thinking about it...

Reply:
But the framework wouldn't know what to do about that version number.